### ISSUE TITLE
GetAsByteArray() leads to exception if columns contain Richtext formatted data

### ISSUE THREAD

**Issue by VorbergHolger** [EPPlusSoftware/EPPlus#858]
In version **6.2.0 and above** the .GetAsByteArray() methods leads to an exception "**Object reference not set to an instance of an object**" if the columns contain Richtext formatted data.

Version 6.1.3 works fine, all newer Versions don't.

**Comment by OssianEPPlus** [comment]
Hia! Tested this when it came in via Freshdesk yesterday. I am unable to reproduce this issue as when I put richText into a cell and then use .GetAsByteArray() it works fine in both 6.2.0 and 6.2.2. Could you send us a reproducible code sample? Perhaps I've missunderstood the type of Richtext formatting you're using.

**Comment by VorbergHolger** [comment]
Hello and thanks for your response.

I’ve attached a small console test solution, created with Visual Studio 2022 on .net Core 6.0 to this email.

It uses an Excel document (RTF_Labels.xlsx) as a template in the same way it is used in my main application.
The error does not occur if you use EPPlus Version 6.1.3 but in each newer version of EPPlus.

Mit freundlichen Grüßen / Best regards

Holger Vorberg

Solution Business Building Automation - Engineering (BT-IE/ENG1-BA)
Bosch Building Automation GmbH | Postfach 2154 | 33404 Verl | GERMANY | www.digicontrol.info<http://www.digicontrol.info/>
***@***.******@***.***>

Sitz: D-33415 Verl, Registergericht: AG Gütersloh HRB 1464, USt.-IdNr. : DE126788839
Geschäftsführung: Volker Westerheide, Jan Wittmann

​
Von: OssianEPPlus ***@***.***>
Gesendet: Dienstag, 25. April 2023 13:41
An: EPPlusSoftware/EPPlus ***@***.***>
Cc: Vorberg Holger (BT-IE/ENG6-BA) ***@***.***>; Author ***@***.***>
Betreff: Re: [EPPlusSoftware/EPPlus] GetAsByteArray() leads to exception if columns contain Richtext formatted data (Issue #858)


Hia! Tested this when it came in via Freshdesk yesterday. I am unable to reproduce this issue as when I put richText into a cell and then use .GetAsByteArray() it works fine in both 6.2.0 and 6.2.2. Could you send us a reproducible code sample? Perhaps I've missunderstood the type of Richtext formatting you're using.

**Comment by OssianEPPlus** [comment]
I can't see any attachments within the E-mail I'm afraid. Something seems to have gone wrong. You should be able to upload it directly in this thread rather than by e-mail

**Comment by VorbergHolger** [comment]
Hello,
your eMail system has probably filtered out the attachment.
How can I upload files to this thread?

**Comment by OssianEPPlus** [comment]
[Like this](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files)

**Comment by VorbergHolger** [comment]
Ok
[EPPlus_Test_RTF.zip](https://github.com/EPPlusSoftware/EPPlus/files/11330414/EPPlus_Test_RTF.zip)

**Comment by OssianEPPlus** [comment]
Do you have a commercial license for Epplus? Would you send me the license number on ossian.edstrom@epplussoftware.com and then we can open a support ticket for you.

**Comment by VorbergHolger** [comment]
Yes, my company has purchased a commercial license for one developer. I'm the only developer, who is working with this Library. I'm pretty sure, that I won't be able to retrieve the license number shortly, but I will give it a try.

**Comment by OssianEPPlus** [comment]
We'll continue this via e-mail

**Comment by OssianEPPlus** [comment]
We'll be resolving your issue in the meantime. We found where the error happens but have yet to determine full cause

**Comment by VorbergHolger** [comment]
Yes, I've got your mail.

**Comment by JanKallman** [comment]
Fixed in EPPlus 6.2.3
